src/mixins/object_geometry.mixin.js

```diff
@@ -565,7 +565,7 @@
      * @return {Object} .y height dimension
      */
     _getNonTransformedDimensions: function() {
-      var strokeWidth = this.strokeWidth,
+      var strokeWidth = this.strokeUniform ? 0 : this.strokeWidth,
```

Comment:
this part does not convince me.
This is to get object dimensions, and the objects dimensions do not change with stroke uniform, before transformation.
So this change, while can be useful fir this case, is not correct for others.

Reply:
Yeah, I also had some reservations about this change. 

The original implementation unfortunately doesn't take stroke scaling into account (since those are "non-transformed dimensions"). I found out that transformed dimension finalization in `_finalizeDimensions` includes the stroke width, but it is after the initial stroke width has been included in the `_getNonTransformedDimensions` calculation. After applying the scaling, the initial `strokeWidth` gets scaled as well in the final dimensions, which is wrong for `strokeUniform: true`.

I'll try to think of a different solution, which doesn't modify the `_getNonTransformedDimensions` function significantly.

Reply:
you can use naked width and height instead of calling _getNonTransformedDimensions, not a big deal `{ x: this.wdith, y: this.height }`

Reply:
You're right, it will be simpler and safer that way :) I changed the code in a new commit.